### PR TITLE
Added GH Actions-based CI.

### PR DESCRIPTION
**Pull request by thresheek**
This is probably not the final version we'd like to have here for the following reasons:
- it's ugly
- it does not cover all supported distros
- it does not cover all support arches - amd64, but not aarch64 (and we probably need debian/arm32v5, too...)
- it's probably way too many jobs and we should test "nginx modules + njs + otel" and "all modules" for each distro/arch
- do we really need schedule?

The current results are available at:
 - "pull request" / "push" : https://github.com/thresheek/pkg-oss/actions/runs/10712504478
 - "schedule" / "workflow_dispatch": https://github.com/thresheek/pkg-oss/actions/runs/10725651712

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+          UBUNTU="$UBUNTU]}"
+
+          if [[ $REDHAT == *, ]]; then
+           REDHAT="${REDHAT%?}"
+          fi
+          REDHAT="$REDHAT]}"
+
+          echo "matrix-alpine=$( echo "$ALPINE" )" >> $GITHUB_OUTPUT
+          echo "matrix-ubuntu=$( echo "$UBUNTU" )" >> $GITHUB_OUTPUT
+          echo "matrix-redhat=$( echo "$REDHAT" )" >> $GITHUB_OUTPUT
```
Comment (oxpa) [inline .github/workflows/ci.yml:72]: I don't quite like building json using echo. It's a PITA to edit this code later. The following snippet does the same and either fail straight away or produce a valid json.
It's also shorter.
```
for os in alpine ubuntu redhat; do
    M=$COMMON_MODULES
    if [ "$os" != "redhat" ]; then
        M="$M $ADD_MODULES"
    fi
    jq -nRr '"matrix-" + $os +"=" + ( { include: [ $ARGS.positional[] | {module: . } ] } | tojson)' --arg os $os --args $M  >> $GITHUB_OUTPUT
done
```

Comment (thresheek) [reply .github/workflows/ci.yml]: Implemented in the current patch version, thanks.

**Comment by thresheek** [comment]
Schedule removed from the current patch version.

**Review by oxpa — APPROVED** [review]

**Comment by thresheek** [comment]
Looks good enough as a first step, let's adapt it further later.

**Pull request merged**